src/etc/inc/services.inc

```diff
@@ -841,7 +844,14 @@ EOPP;
 				$dhcpdconf .= "		$deny_action dynamic bootp clients;\n";
 			}
 
-			if (isset($poolconf['denyunknown'])) {
+			// set pool MAC limitations
+			if ($poolconf['denyunknown'] == "class") {
+				$dhcpdconf .= "		allow members of \"s_{$dhcpif}\";\n";
+ 			//	$dhcpdconf .= "		$deny_action known-clients;\n";        // in combination with above and below options, this will disable all entries WITHOUT FIXED IPs!
+				$dhcpdconf .= "		$deny_action unknown-clients;\n";
+			} else if ($poolconf['denyunknown'] == "disabled") {
+				// add nothing to $dhcpdconf; condition added to prevent next condition applying if ever engine changes such that: isset("disabled") == true
+			} else if (isset($poolconf['denyunknown'])) {	// "catch-all" covering "enabled" value post-PR#4066, and covering non-upgraded boolean option (i.e. literal value "enabled")
 				$dhcpdconf .= "		$deny_action unknown-clients;\n";
```

Comment:
It doesn't make much sense to check if `$poolconf['denyunknown']` is set at this point. IMO this whole block should be simplified using something like:
```
if (isset($poolconf['denyunknown']) &&
    $poolconf['denyunknown'] != "disabled") {
	if ($poolconf['denyunknown'] == "class") {
		$dhcpdconf .= "		allow members of \"s_{$dhcpif}\";\n";
	}
	$dhcpdconf .= "		$deny_action unknown-clients;\n";
}
```

Reply:
Apologies for the delay. I disagree:

1. the `isset` is to check for backwards (as-yet-not-upgraded) `denyunknown` values
2. without considering _frequency_, the universe of `!= "disabled"` (negative tests) is far larger than positive tests, and would also be clearer and more concise if/when future values are added